beacon_node/network/src/sync/range_sync/chain.rs

```diff
@@ -689,16 +699,29 @@ impl<T: BeaconChainTypes> SyncingChain<T> {
             return None;
         }
 
+        let batch_start_slot = std::cmp::max(
```

Comment:
I think most batches SHOULD start on an epoch boundary. In the case that they don't (for whatever reason) we should make sure we don't do requests that could miss any blocks. For example, if an epoch is 5 blocks and `self.start_slot` is 12.. we would want to download from 10-15. 
If i'm not mistaken, in this case, the logic here would start from 15? And we would miss blocks 12-15, in doing so the sync would fail because we couldn't process the earlier blocks. 

It might make more sense to ensure the `chain.start_slot` aligns on an epoch boundary, and leave the `get_next_batch()` as is. As you've set the blocks to be epoch's per batch we should always align (fingers crossed). 

The alternative, is to keep the logic you have here, but favour the lower epoch, so at worse we re-download blocks. The issue here being that each time we process `get_next_batch` the calculation keeps re-aligning. If we sanitised `self.start_slot` every time it gets modified to an epoch boundary, this should also force `get_next_batch` to always align without having to check it.

Reply:
`to_be_downloaded` starts at 1, so the second part of this comparison would be:
```rust
(self.start_slot + self.to_be_downloaded_id.saturating_sub(1) * blocks_per_batch)
                .epoch(slots_per_epoch)
                .start_slot(slots_per_epoch),
```
which is `10`, so the first batch would start in 12

Reply:
I thought about ensuring `start_slot` is always aligned to an epoch (and except a couple of lines - i think - whatever is feed to this field is indeed aligned), but then this leaves the need to remember to align what goes to `start_slot` everytime it's changed. I thought changing `get_next_batch` was more future-proof.

About things being per epochs, this is a change in the middle: it keeps batches' logic in the slot realm but tackling the change of `slots_per_epoch` issue when changing specs. I did it this way to avoid re-downloading those first known slots, in your example from 10 to 12.

So, I could either move everything to be epochs (in that case this part doesn't change much since it now wouldn't have the last `start_slot()` or search that couple of lines where I think the `start_slot` is not guaranteed to be aligned and force the alignment. Maybe to address my future-proofing concern, document the `start_slot` field as something that should be an epoch boundary?

Reply:
Late comment: if `start_slot` is expected to be an epoch boundary, what about making it an `epoch`? Then there is no needed effort on remembering it must stay aligned if/when the code changes

Reply:
Yep you're right, I missed the 0 case. This PR looks good. 

I like the idea of your late comment though. I think we should make start slot a `start_epoch` to simplify the logic. If you have the bandwidth I think this would be a great modification.

Reply:
Sure I'll do it

Reply:
After checking some more, I realize that `start_slot` is not always supposed to align to an epoch boundary since the `SyncingChain` is used both for finalized chains and head chains. 
If I understand correctly, moving to a `start_epoch` would essentially collapse finalized chains and head chains, or at least collapse a good amount of head chains together, and would also bring a lot more of already known blocks to play